test/markdown_test.dart

```diff
@@ -716,6 +722,40 @@ void main() {
         ''');
   });
 
+  group('Inline HTML', () {
+    validate(
+        'within a paragraph',
+        '''
+        Within a <em class="x">paragraph</EM>.
+        ''',
+        '''
+        <p>Within a <em class="x">paragraph</EM>.</p>
+        ''',
+        extensions: ['inline-html']);
+
+    validate(
+        'not HTML',
+        '''
+        Obviously, 3 < 5 and 7 > 2.
+        Not HTML: <3>, <_a>, <>
+        ''',
+        '''
+        <p>Obviously, 3 &lt; 5 and 7 > 2.
+        Not HTML: &lt;3>, &lt;_a>, &lt;></p>
+        ''',
+        extensions: ['inline-html']);
+
+    validate(
+        '"markdown" within a tag is not parsed',
+        '''
+        Text <a href="_foo_">And "_foo_"</a>.
+        ''',
+        '''
+        <p>Text <a href="_foo_">And "<em>foo</em>"</a>.</p>
+        ''',
+        extensions: ['inline-html']);
+  });
+
```

Comment:
I think this will all have to be rebased after I merged in @kevmoo's change that moves these tests out to data files. You'll probably want some way to specify which extensions a test data file requires.